stdlib/ballerina-database/src/main/java/org/ballerinalang/database/sql/SQLDataIterator.java

```diff
@@ -255,14 +437,54 @@ public BStruct generateNext() {
                     }
                 }
             }
-        }  catch (Throwable e) {
+        } catch (Throwable e) {
             throw new BallerinaException(
                     "error in retrieving next value for column: " + columnName + ": at index:" + index + ":" + e
                             .getMessage());
         }
         return bStruct;
     }
 
+    private void handleUnionTypeStructField(BStruct bStruct, BStructType.StructField[] structFields, int index,
+            int refRegIndex, BRefType value, int refType) {
+        if (isNillableType(structFields, index - 1, refType)) {
+            bStruct.setRefField(refRegIndex, value);
+        } else {
+            throw new BallerinaException("Union type contains more than 2 members or the members do not result in "
+                    + "an assignable nillable type");
+        }
+    }
+
+    private void handleNilToNonNilableFieldAssignment() {
+        throw new BallerinaException("Trying to assign a Nil value to a non-nillable field");
```

Comment:
Shall we add more information to this error message like field index and column name?